File: kotlin/amen-client/src/main/kotlin/sa/amnn/kit/resources/Lookups.kt
package sa.amnn.kit.resources

import kotlinx.serialization.builtins.ListSerializer
import sa.amnn.kit.AmenClient
import sa.amnn.kit.Lookup

class Lookups(private val c: AmenClient) {
    private val list = ListSerializer(Lookup.serializer())
    suspend fun cities() = c.request("GET", "/cities", list) ?: emptyList()
    suspend fun categories() = c.request("GET", "/categories/", list) ?: emptyList()
    suspend fun disputeReasons() = c.request("GET", "/dispute-reasons/", list) ?: emptyList()
    suspend fun disputeResolutionReasons() = c.request("GET", "/dispute-resolution-reasons/", list) ?: emptyList()
    suspend fun cancelReasons(partyType: String? = null) = c.request("GET", "/cancel-reasons/", list, mapOf("party_type" to partyType)) ?: emptyList()
}

File: swift/Sources/AmenClient/Resources/Lookups.swift
public final class Lookups {
    let c: AmenClient; init(_ c: AmenClient) { self.c = c }
    public func cities() async throws -> [Lookup] { try await c.request(.GET, "/cities") }
    public func categories() async throws -> [Lookup] { try await c.request(.GET, "/categories/") }
    public func disputeReasons() async throws -> [Lookup] { try await c.request(.GET, "/dispute-reasons/") }
    public func disputeResolutionReasons() async throws -> [Lookup] { try await c.request(.GET, "/dispute-resolution-reasons/") }
    public func cancelReasons(partyType: String? = nil) async throws -> [Lookup] { try await c.request(.GET, "/cancel-reasons/", params: ["party_type": partyType]) }
}
